src/EnergyPlus/HVACDXHeatPumpSystem.cc

```diff
@@ -182,38 +182,35 @@ namespace HVACDXHeatPumpSystem {
         // simulate DX Heating System
         CompName = DXHeatPumpSystem(DXSystemNum).HeatPumpCoilName;
 
-        {
-            auto const SELECT_CASE_var(DXHeatPumpSystem(DXSystemNum).HeatPumpCoilType_Num);
-
-            if (SELECT_CASE_var == CoilDX_HeatingEmpirical) { // COIL:DX:COOLINGBYPASSFACTOREMPIRICAL
-
-                SimDXCoil(state,
-                          CompName,
-                          CompressorOperation::On,
-                          FirstHVACIteration,
-                          DXHeatPumpSystem(DXSystemNum).HeatPumpCoilIndex,
-                          DXHeatPumpSystem(DXSystemNum).FanOpMode,
-                          DXHeatPumpSystem(DXSystemNum).PartLoadFrac);
-
-            } else if (SELECT_CASE_var == Coil_HeatingAirToAirVariableSpeed) { // Coil:Heating:DX:VariableSpeed
-                SimVariableSpeedCoils(state,
-                                      CompName,
-                                      DXHeatPumpSystem(DXSystemNum).HeatPumpCoilIndex,
-                                      DXHeatPumpSystem(DXSystemNum).FanOpMode,
-                                      state.dataHVACDXHeatPumpSys->MaxONOFFCyclesperHour,
-                                      state.dataHVACDXHeatPumpSys->HPTimeConstant,
-                                      state.dataHVACDXHeatPumpSys->FanDelayTime,
-                                      CompressorOperation::On,
-                                      DXHeatPumpSystem(DXSystemNum).PartLoadFrac,
-                                      DXHeatPumpSystem(DXSystemNum).SpeedNum,
-                                      DXHeatPumpSystem(DXSystemNum).SpeedRatio,
-                                      state.dataHVACDXHeatPumpSys->QZnReq,
-                                      state.dataHVACDXHeatPumpSys->QLatReq,
-                                      state.dataHVACDXHeatPumpSys->OnOffAirFlowRatio);
-
-            } else {
-                ShowFatalError(state, "SimDXCoolingSystem: Invalid DX Heating System/Coil=" + DXHeatPumpSystem(DXSystemNum).HeatPumpCoilType);
-            }
+        switch (DXHeatPumpSystem(DXSystemNum).HeatPumpCoilType_Num) {
+        case CoilDX_HeatingEmpirical: { // COIL:DX:COOLINGBYPASSFACTOREMPIRICAL
```

Comment:
Is it worth doing the `auto & dxHeatPumpSys = state.dataDXHeatPumpSystem->DXHeatPumpSystem(DXSystemNum);` thing here? Probably not at this point, but :pushpin:.